zio-http/shared/src/main/scala/zio/http/URL.scala

```diff
@@ -279,26 +280,35 @@ final case class URL(
 }
 
 object URL {
-  def empty: URL = URL(Path.empty)
+  val empty: URL = URL(path = Path.empty)
+
+  /**
+   * Proper implementation of [[java.net.MalformedURLException]] which, unlike
+   * the Java version, can propagate the cause.
+   */
+  final case class MalformedURLException(rawUrl: String, cause: Option[Throwable])
```

Comment:
I tend to write `cause: Throwable = null` instead of using an `Option` here but Scala people tend to get shocked when `null` is used anywhere.

@987Nabil @kyri-petrou What do you prefer?